schema/profile.js

```diff
@@ -26,6 +28,67 @@ module.exports = db => {
     }
   });
 
+  Profile.searchAndCountAll = ({ search, ...rest }) => {
+
+    const searchFullName = search => {
+      search = search.split(' ');
+      if (search.length > 1) {
+        return {
+          [Op.and]: [
+            { firstName: { [Op.iLike]: `%${search[0]}` } },
+            { lastName: { [Op.iLike]: `${search[1]}%` } }
+          ]
+        };
+      }
+      return {
+        [Op.or]: [
+          { firstName: { [Op.iLike]: `%${search[0]}%` } },
+          { lastName: { [Op.iLike]: `%${search[0]}%` } }
+        ]
+      };
+    };
+
+    const where = search
+      ? {
+        [Op.or]: [
+          searchFullName(search),
+          { '$pil.licenceNumber$': { [Op.iLike]: `%${search}%` } },
+          // cast ENUM field to TEXT for search
+          db.where(db.cast(db.col('type'), 'TEXT'), { [Op.eq]: search })
+        ]
+      }
+      : {};
+
+    return Promise.all([
+      Profile.count(),
```

Comment:
This only works because we only have one establishment with data. It will need to filter to the establishment.

Reply:
Aha, I'm glad you brought that up...

That's in the API, as it uses req.establishment.id. the where clauses get merged

Reply:
now _that_ is a very good point. 🎩